addons/mail/models/mail_activity.py

```diff
@@ -366,13 +367,29 @@ def unlink(self):
 
     @api.model
     def _search(self, domain, offset=0, limit=None, order=None, *, bypass_access=False, **kwargs):
-        """ Override that adds specific access rights of mail.activity, to remove
-        ids uid could not see according to our custom rules. Please refer to
-        :meth:`_check_access` for more details about those rules.
+        """ Override _search to implement custom access rules and search behavior.
 
-        The method is inspired by what has been done on mail.message. """
+        This method enhances the standard search in two ways:
+
+        1.  **Custom Access Rights**: It filters search results to only include
+            activities the current user is allowed to see. An activity is
+            accessible if the user is the assignee (`user_id`), or if they have
+            read access to the related document (`res_model`, `res_id`). This
+            logic is detailed in :meth:`~._check_access`.
+
+        2.  **Archived Records Search**: If the search domain includes a filter on
+            the 'Done Date' (`date_done`) field, the search automatically includes
+            archived records (``active_test=False``). This allows users to find
+            past activities that have been completed and subsequently archived.
+
+        Superusers bypass these custom rules and perform a standard search.
+        """
 
         # Rules do not apply to administrator
+
+        if any(condition.field_expr == 'date_done' for condition in Domain(domain).iter_conditions()):
+            kwargs['active_test'] = False
+
```

Comment:
```suggestion

        if any(condition.field_expr == 'date_done' for condition in Domain(domain).iter_conditions()):
            kwargs['active_test'] = False

```

Would it be more readable to do this instead?

```
        active_test = kwargs.pop('active_test', True)
        if any(condition.field_expr == 'date_done' for condition in Domain(domain).iter_conditions()):
            active_test = False

        if self.env.is_superuser() or bypass_access:
            return super()._search(domain, offset, limit, order, active_test=active_test, bypass_access=True, **kwargs)

        # retrieve activities and their corresponding res_model, res_id
        # Don't use the ORM to avoid cache pollution
        query = super()._search(domain, offset, limit, order, active_test=active_test, **kwargs)
```